fsm.cpp

```diff
@@ -18,12 +20,29 @@ int state1_countdown_timer = 0;
 int state2_duration_timer = 0;
 int state3_stillness_timer = 0;
 
+std::queue<sensorData_t> sensorDataQueue;
```

Comment:
I'm still trying to work through what I'd recommend, but take a look at this article on a threadsafe queue: https://codetrips.com/2020/07/26/modern-c-writing-a-thread-safe-queue/ . I think the person does a good job of articulating what needs to be done for sharing a queue across different threads.

Reply:
What I'm not sure about is whether these interrupts run their code in a new thread or not.

Reply:
As discussed, I was unable to. I have added a note.